doc/zmq_poller.txt

```diff
@@ -0,0 +1,269 @@
+zmq_poller(3)
+===========
+
+
+NAME
+----
+zmq_poller - input/output multiplexing
+
+
+SYNOPSIS
+--------
+
+*void *zmq_poller_new (void);*
+*int zmq_poller_destroy (void **'poller_p');*
+
+*int zmq_poller_add (void *'poller', void *'socket', void *'user_data', short 'events');
+*int zmq_poller_modify (void *'poller', void *'socket', short 'events');*
+*int zmq_poller_remove (void *'poller', void *'socket');*
+
+*int zmq_poller_add_fd (void *'poller', int 'fd', void *'user_data', short 'events');*
+*int zmq_poller_modify_fd (void *'poller', int 'fd', short 'events');*
+*int zmq_poller_remove_fd (void *'poller', int 'fd');*
+
+*int zmq_poller_wait_all (void *'poller',
+                          zmq_poller_event_t *'events',
+                          int 'n_events',
+                          long 'timeout');*
+
+DESCRIPTION
+-----------
+The _zmq_poller_*_ functions provide a mechanism for applications to multiplex
+input/output events in a level-triggered fashion over a set of sockets.
+
+_zmq_poller_new_ and _zmq_poller_destroy_ manage the lifetime of a poller
+instance. _zmq_poller_new_ creates and returns a new poller instance, while
+_zmq_poller_destroy_ destroys it. A pointer to a valid poller must be passed
+as the _poller_p_ argument of _zmq_poller_destroy_. In particular,
+_zmq_poller_destroy_ may not be called multiple times for the same poller
+instance. _zmq_poller_destroy_ sets the passed pointer to NULL in case of a
+successful execution. _zmq_poller_destroy_ implicitly unregisters all
+registered sockets and file descriptors.
+
+_zmq_poller_add_, _zmq_poller_modify_ and _zmq_poller_remove_ manage the 0MQ
+sockets registered with a poller.
+
+_zmq_poller_add_ registers a new _socket_ with a given _poller_. Both _poller_
+and _socket_ must point to valid 0MQ objects. The _events_ parameter specifies
+which event types the client wants to subscribe to. It is legal to specify no
+events (i.e. 0), and activate them later with _zmq_poller_modify_.
+In addition, _user_data_ may be specified, which is not used by the poller, but
+passed back to the caller when an event was signalled in a call to
+_zmq_poller_wait_ or _zmq_poller_wait_all_. _user_data_ may be NULL. If it is
+not NULL, it must be a valid pointer. Otherwise, behaviour is undefined.
```

Comment:
Just minor suggestion and feel free to ignore. Is it not up to the user to decide the content of a pointer and if this is undefined behavior or not? For example someone might decide to use it for passing index values etc..

Reply:
That's the motivation for the remark. Someone might have this idea, but this explains that they should drop it ;) The argument type is a pointer, and you cannot legally use a pointer to transport a non-pointer.

One might argue that this remark is superfluous as this is illegal already due to type aliasing rules, but since it might be a typical idea of misuse, it is good to have it here.